src/System.Diagnostics.Process/src/System/Diagnostics/Process.Windows.cs

```diff
@@ -445,7 +445,16 @@ private bool StartWithCreateProcess(ProcessStartInfo startInfo)
             //    * CreateProcess allows you to redirect all or none of the standard IO handles, so we use
             //      GetStdHandle for the handles that are not being redirected
 
-            StringBuilder commandLine = BuildCommandLine(startInfo.FileName, startInfo.Arguments);
+            string arguments;
+            if (startInfo.ArgumentList.Count > 0)
+            {
+                arguments = PasteArguments.Paste(startInfo.ArgumentList, pasteFirstArgumentUsingArgV0Rules: false);
+            }
+            else
+            {
+                arguments = startInfo.Arguments;
+            }
+            StringBuilder commandLine = BuildCommandLine(startInfo.FileName, arguments);
```

Comment:
If ArgumentList.Count > 0, we're going to create a string builder and use it to create a string for the arguments, only to then turn around and create another one to create another string for the command-line.  Can we somehow combine those, or reutilize the original StringBuilder, rather than generating extra garbage?